department-of-veterans-affairs/vets-website
# Add submit transformer

## Summary
Add 'submit transformer' in the Pension Benefits application(21P-527EZ) to remove stale form data before submission

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Sign in via mocked authentication at `http://localhost:3001/sign-in/mocked-auth`
3. Copy and paste 'maximal-test.json' into the 'Save in Progress' menu
4. Set return url to '/review-and-submit' in the 'Save in Progress' menu
5. Go to the 'Review application' page (/review-and-submit)
6. Edit a conditional field that should remove stale data (See test case for example)
8. Verify that all chapters and pages are validated before submitting the form
9. Verify that the payload form data in the `/v0/pension_claims` api post network request is transformed correctly based on any conditional properties

## Test Case
1. Load 'maximal-test.json' into the application
2. On the 'Review application' expand the 'Household Information' chapter
3. Edit 'Marital Status' to 'Divorced'
4. Edit 'Dependents' to 'No'
5. Verify that all 'Marriages' and 'Dependents' sections are removed from the UI
6. Submit the form and validate

## Related issue(s)
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73036)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

## Screenshots
n/a


## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73036)

### Quality Assurance & Testing

### Error Handling


### Authentication
